packages/flame/lib/src/components/sprite_component.dart

```diff
@@ -30,9 +64,14 @@ class SpriteComponent extends PositionComponent
     super.anchor,
     super.children,
     super.priority,
-  }) : super(
-          size: size ?? sprite?.srcSize,
-        ) {
+  })  : assert(
+          (size == null && (autoResize ?? true)) ||
+              (size != null && !(autoResize ?? false)),
```

Comment:
```suggestion
          (size == null && (autoResize ?? true)) ||
              (size != null && !(autoResize ?? false)),
```
Hmm, shouldn't this be possible to write as a single XOR expression?

Reply:
Simplified as suggested.